core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -88,31 +88,31 @@ private DataTask task(StaticTableScan scan) {
     return StaticDataTask.Row.of(partition.key, partition.recordCount, partition.fileCount);
   }
 
-  private Iterable<Partition> partitions(StaticTableScan scan) {
+  private static Iterable<Partition> partitions(StaticTableScan scan) {
```

Comment:
This could continue to be static as long as we get all of the data correctly from `scan`. The original table instance can be accessed through `scan.table()` and the partition table schema is `scan.schema()`.